### DISCUSSION TITLE
How do I specify a base model when training? (spacy3)

### DISCUSSION THREAD

**Discussion by erotavlas** [explosion/spaCy#7766] [Help: Coding & Implementations] [answered]
I created a base model with custom tokenization.  How do I use this to train a new model so that the tokenization rules get transferred to the trained model? In spacy 2, I used `--base-model` flag in the command line.   

See here for more details on how I added the tokenization rules https://github.com/explosion/spaCy/issues/5241#issue-592184132

**EDIT**:  Is it correct that this is the way to do it - using custom functions? https://spacy.io/usage/training#custom-functions
If so, does the name of the function matter?  Also will these changes become part of the model itself, or do I have to save a pipeline?  I would prefer to keep the rules embedded in the model like before if possible.  I think that's what its doing based on the documentation but just wanted to verify.

Thanks

**Comment by adrianeboyd** [comment, marked as answer]
Yes, see the example here: https://spacy.io/usage/training#custom-tokenizer 

And if you want to copy the tokenizer from an existing model (more like the "base model" approach), see the discussion in #7582.

The custom tokenizer settings are saved with the model, so you only need to do this once on init. If you have the code at hand to create the custom settings, then your own `before_init` callback is probably easiest. You have to provide it with `spacy train --code` but only once on for the initialization step, so others loading the saved/packaged model in the future won't need the custom code. (The custom callback is recorded in the config and you can provide it if you want with `spacy package --code` to make it easier for others to replicate the training setup in the future, but it's not required.)
